satellite/satellitedb/overlaycache.go

```diff
@@ -1338,3 +1338,121 @@ func populateUpdateFields(dbNode *dbx.Node, updateReq *overlay.UpdateRequest) db
+		value := pb.Node{
+			Id: node.NodeID,
+			Address: &pb.NodeAddress{
+				Transport: pb.NodeTransport_TCP_TLS_GRPC,
+				Address:   node.Address.GetAddress(),
+			},
+		}
+		err := cache.UpdateAddress(ctx, &value, config)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		_, err = cache.UpdateUptime(ctx, node.NodeID, node.IsUp, config.UptimeReputationLambda, config.UptimeReputationWeight, config.UptimeReputationDQ)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		pbInfo := pb.InfoResponse{
+			Operator: node.Operator,
+			Capacity: node.Capacity,
+			Type:     pb.NodeType_STORAGE,
+		}
+		_, err = cache.UpdateNodeInfo(ctx, node.NodeID, &pbInfo)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+	case *pq.Driver:
+		// v is a single feedback value that allows us to update both alpha and beta
+		var v float64 = -1
+		if node.IsUp {
+			v = 1
+		}
+		uptimeReputationAlpha := config.UptimeReputationLambda*config.UptimeReputationAlpha0 + config.UptimeReputationWeight*(1+v)/2
+		uptimeReputationBeta := config.UptimeReputationLambda*config.UptimeReputationBeta0 + config.UptimeReputationWeight*(1-v)/2
+		start := time.Now()
+		query := `
+			INSERT INTO nodes
+			(
+				id, address, last_net, protocol, type,
+				email, wallet, free_bandwidth, free_disk,
+				uptime_success_count, total_uptime_count, 
+				last_contact_success,
+				last_contact_failure,
+				audit_reputation_alpha, audit_reputation_beta, uptime_reputation_alpha, uptime_reputation_beta
+			)
+			VALUES (
+				$1, $2, $3, $4, $5,
+				$6, $7, $8, $9,
+				$10::bool::int, 1,
+				CASE WHEN $10 IS TRUE THEN current_timestamp
+					ELSE '0001-01-01 00:00:00+00'
+				END,
+				CASE WHEN $10 IS FALSE THEN current_timestamp
+					ELSE '0001-01-01 00:00:00+00'
+				END,
+				$11, $12, $13, $14
+			)
+			ON CONFLICT (id)
+			DO UPDATE
+			SET
+				address=$2,
+				last_net=$3,
+				email=$6,
+				wallet=$7,
+				free_bandwidth=$8,
+				free_disk=$9,
+				total_uptime_count=nodes.total_uptime_count+1,
+				uptime_reputation_alpha=$16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int,
+				uptime_reputation_beta=$16::numeric*nodes.uptime_reputation_beta + $17::numeric*(NOT $10)::bool::int,
+				uptime_success_count = nodes.uptime_success_count + $10::bool::int,
+				last_contact_success = CASE WHEN $10 IS TRUE
+					THEN current_timestamp
+					ELSE nodes.last_contact_success
+				END,
+				last_contact_failure = CASE WHEN $10 IS FALSE
+					THEN current_timestamp
+					ELSE nodes.last_contact_failure
+				END,
+				-- this disqualified case statement resolves to: 
+				-- when (new.uptime_reputation_alpha /(new.uptime_reputation_alpha + new.uptime_reputation_beta)) <= config.UptimeReputationDQ
+				disqualified = CASE WHEN (($16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int) / (($16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int) + ($16::numeric*nodes.uptime_reputation_beta + $17::numeric*(NOT $10)::bool::int))) <= $15
```

Comment:
Here’s a potential problem- if the node meets the disqualification criteria but it is already disqualified, we are still updating the disqualification timestamp here. It might be difficult then to go back and see when a node was _first_ disqualified. A node operator might even take advantage of this on purpose, so they could tell the support team “but it just got DQ’d today!” when really it has been out for a month.

The solution probably has to be something like `disqualified = CASE WHEN (huge existing expression) <= $15 AND NOT nodes.disqualified`

Reply:
ah great catch! let me fix that